contracts/jstz_fa_bridge.mligo

```diff
@@ -0,0 +1,127 @@
+#import "./common/entrypoints/ticketer-deposit.mligo" "TicketerDepositEntry"
+#import "./common/entrypoints/router-withdraw.mligo" "RouterWithdrawEntry"
+#import "./common/tokens/tokens.mligo" "Token"
+#import "./common/types/ticket.mligo" "Ticket"
+#import "./common/errors.mligo" "Errors"
+#import "./common/assertions.mligo" "Assertions"
+
+#include "./common/jstz_type.mligo"
+
+module Storage = struct
+  (* Stores the receiver and rollup address during a bridging process. 
+     It is set when deposit entrypoint is executed and cleared after 
+     the ticket is received from the ticketer. *)
+  type ongoing_deposit =
+    {
+    receiver : address;
+      (* Address that will receive tokens in Jstz *)
+    rollup : address
+      (* Address of the Jstz smart rollup contract *)
+    }
+
+  type t =
+    { 
+      token : Token.t;
+        (* Token which Ticketer accepts for minting tickets, immutable *)
+      ticketer : address;
+        (* Ticketer address, immutable *)
+      proxy : address option;
+        (* Proxy smart function address, immutable *)
+      ongoing_deposit : ongoing_deposit option;
+        (* Details of the on-going deposit *)
+      metadata : (string, bytes) big_map
+        (* Metadata of the contract (TZIP-016), immutable *)
+    }
+
+  let set_ongoing_deposit (ongoing_deposit: ongoing_deposit) (store : t) : t =
+    {store with ongoing_deposit = (Some ongoing_deposit)}
+
+  let clear_ongoing_deposit (store : t) : t = {store with ongoing_deposit = None}
+end
+
+module Jstz_fa_bridge = struct
+  module Tezos = Tezos.Next
+
+  type storage = Storage.t
+
+  type return = operation list * storage
```

Comment:
Note to self: change LIGO type inference for contracts to avoid requiring a `return` type annotation